Cpix/DrmSystem.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.IO;
+using System.Xml;
+using System.Xml.XPath;
+
+namespace Axinom.Cpix
+{
+	public sealed class DrmSystem : Entity
+	{
+		/// <summary>
+		/// Gets or sets the unique ID of the DRM system. Refer to the DASH-IF System
+		/// ID registry for a list of DRM system IDs.
+		/// </summary>
+		public Guid Id { get; set; }
+
+		/// <summary>
+		/// Gets or sets the ID of the content key the DRM system references.
+		/// </summary>
+		public Guid KeyId { get; set; }
+
+		/// <summary>
+		/// Gets or sets the content protection data, which is the full well-formed
+		/// XML fragment that shall be added under the ContentProtection element in a
+		/// DASH manifest. This must be a UTF-8 XML string without a byte order mark,
+		/// encoded as base64.
+		/// </summary>
+		public string ContentProtectionData { get; set; }
+
+		internal override void ValidateNewEntity(CpixDocument document)
+		{
+			ValidateEntity();
+		}
+
+		internal override void ValidateLoadedEntity(CpixDocument document)
+		{
+			ValidateEntity();
+		}
+
+		private void ValidateEntity()
+		{
+			if (Id == Guid.Empty)
+				throw new InvalidCpixDataException("An ID must be provided for each DRM system.");
```

Comment:
While the element is called "DRMSystem", this is not "a DRM system". The DRM system is a dll file on the client device. This is just the signaling that associates a DRM system with a content key.

As such, it should not be called "DRM system". Something like "DRM system signaling entry" would be better.

It then follows that "an ID" is rather vague. "The DRM system ID" would be better.

See also all other messages that use this term.

Reply:
Will use "DRM system signaling entry" for descriptions. But kept variables as "DrmSystem". Not sure if going for long e.g. "DrmSystemSignalingEntries" as the collection name is better. "DrmSystem" also nicely aligns with how it is in the spec.